controllers/domain/getDomain.js

```diff
@@ -68,7 +68,11 @@ module.exports = async (req, res) => {
             }
           }
 
-          if (item?.domain?.credderScore >= CREDDER_MIN_SCORE || !CREDDER_CHECK_ENABLED) {
+          const isNewsExist = data.find((news) => news.content.news_url === item.content.news_url);
+          if (
+            (item?.domain?.credderScore >= CREDDER_MIN_SCORE || !CREDDER_CHECK_ENABLED) &&
+            !isNewsExist
+          ) {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The current implementation uses `Array.prototype.find` to check if a news item already exists in the data array. This operation has a time complexity of O(n), which can be inefficient when dealing with large arrays. Consider using a `Set` or an `Object` to store the URLs, which would allow for constant-time lookup operations and improve performance.
```diff
- const isNewsExist = data.find((news) => news.content.news_url === item.content.news_url);
+ let urlSet = new Set(data.map(news => news.content.news_url));
+ const isNewsExist = urlSet.has(item.content.news_url);
```
Please note that you need to initialize `urlSet` before the loop where this code resides. If it's inside a loop, move it outside to avoid reinitializing it on every iteration.


<!-- This is an auto-generated comment by OSS CodeRabbit -->